src/scorecard.js

```diff
@@ -0,0 +1,28 @@
+class ScoreCard {
+  constructor(){
+    this.rolls = [];
+  };
+
+  roll(nPins){
+    this.rolls.push(nPins);
+  };
+
+  score(){
+    let s = 0;
+    let r = 0;
+    for (let frames = 0; frames < 10; frames++){
+      if(this.rolls[r] === 10){
+        s += this.rolls[r] + this.rolls[r+1] + this.rolls[r+2]
+        r += 1;
+      } else if (this.rolls[r] + this.rolls[r+1] === 10){
+        s += this.rolls[r] + this.rolls[r+1] + this.rolls[r+2]
+        r += 2
+      } else {
+        s += this.rolls[r] + this.rolls[r+1]
+        r += 2
+      };
+    };
+    return s;
```

Comment:
Great work implementing the logic for strikes and spares, I didn't get anywhere close. As discussed given more time this would be reformatted into multiple methods to adhere to SRP. I guess first step would be to ship the code executed in each of the if and else blocks into separate methods, so it would read like 'if (condition) { return methodName }'